pkg/apis/resource/validation/validation.go

```diff
@@ -675,19 +685,12 @@ func validateDeviceAttribute(attribute resource.DeviceAttribute, fldPath *field.
 		numFields++
 	}
 	if attribute.StringValue != nil {
-		if len(*attribute.StringValue) > resource.DeviceAttributeMaxValueLength {
-			allErrs = append(allErrs, field.TooLong(fldPath.Child("string"), "" /*unused*/, resource.DeviceAttributeMaxValueLength))
-		}
```

Comment:
This looks like a bad rebase or saving an out-dated file on top of the more recent one. I did not intend to change this... will revert.

Reply:
Now I remember... this came from admin-controlled attributes. Splitting out the code into separate functions made it reusable. Probably a worthwhile enhancement by itself, even if not directly required anymore.